lightprop/io/io.py

```diff
@@ -1,5 +1,93 @@
-def load(file, **kwargs):
-    pass
+import numpy as np
+from PIL import Image
+
+from lightprop.lightfield import LightField
+
+
+def load(
+    *, full_file: str = None, re_file: str = None, im_file: str = None, amp_file: str = None, phase_file: str = None
+) -> LightField:
+    """
+    Loads given files into LightField data structure. Either full_file, any single field file, or a pair of
+    complementary fields is a valid argument combination. full_file must be of .lf type, field files should be images
+    (any format supported by PIL library).
+
+    Example usage:
+
+    >>> lf = load(full_file="holo.lf")
+
+    >>> lf = load(re_file="real.png")
+
+    >>> lf = load(amp_file="amplitude.png", phase_file="/some/directory/phase.bmp")
+
+    Invalid usages:
+
+    >>> lf = load(re_file="real.png", phase_file="/some/directory/phase.bmp")
+
+    :param full_file: path to the custom file (.lf) with full field info
+    :param re_file: path to the image with REAL part of complex field
+    :param im_file: path to the image with IMAGINARY part of complex field
+    :param amp_file: path to the image with AMPLITUDE information of complex field
+    :param phase_file: path to the image with PHASE information of complex field
+    :return: fully initialized LightField instance
+    """
+    # load file with full field data
+    if full_file:
+        return __load_full(full_file)
+
+    # get only non-None arguments
+    fields = {
+        k: v
+        for k, v in {"re": re_file, "im": im_file, "amplitude": amp_file, "phase": phase_file}.items()
+        if v is not None
+    }
+
+    # single field - simple, load and return as LightField
+    if len(fields) == 1:
+        field_name, field_file = [(k, v) for k, v in fields.items() if v is not None][0]
+        return __load_field(field_name, field_file)
+
+    # complementary fields
+    elif len(fields) == 2:
+        if re_file and im_file:
+            cplx = __load_image(re_file) + 1j * __load_image(im_file)
+            amp = np.abs(cplx)
+            phase = np.angle(cplx)
+            return LightField(amp, phase)
```

Comment:
this can be one branch up

Reply:
I mean same return for both branches -> can be extracted

Reply:
Not sure if `if` sequence will exit properly. After all len can be 2 but fields might not be complementary which is an error